crates/ty_python_semantic/resources/mdtest/scopes/nonlocal.md

```diff
@@ -43,3 +43,272 @@ def f():
         def h():
             reveal_type(x)  # revealed: Unknown | Literal[1]
 ```
+
+## The `nonlocal` keyword
+
+Without the `nonlocal` keyword, `x += 1` is not allowed in an inner scope, even if we break it up
+into multiple steps. Local variable scoping is "foward-looking" in the sense that even a _later_
+assignment of `x` means that _all_ reads of `x` in that scope only look at that scope's binding:
+
+```py
+def f():
+    x = 1
+    def g():
+        x += 1  # error: [unresolved-reference]
+
+def f():
+    x = 1
+    def g():
+        y = x  # error: [unresolved-reference]
+        x = y + 1
+```
+
+With `nonlocal` these examples work:
```

Comment:
```suggestion
With `nonlocal` these examples work, because the reassignments modify the variable from the outer scope rather than modifying a separate variable local to the inner scope.
```